pyniva/request_dataframe.py

```diff
@@ -109,3 +111,136 @@ def get_ship_data(
+        return df
+    df = df.sort_values(by="time", ascending=True)
+    df = df.reset_index()
+    df = df.pivot(index = "time", columns="wl", values="value") 
+
+    # then get gps data
+    print(f"Downloading gps track")
+    path = f"{vessel_name}/gpstrack"
+    gps_index = vessel_paths.index(path)
+    gps_data = vessel_signals[gps_index].get_tseries(
+                    pub_tsb,
+                    header=header,
+                    noqc=noqc,
+                    dt=dt,
+                    start_time=start_time,
+                    end_time=end_time,
+    )
+
+    # inperpolate gps data to match ramses
+    print(f"Interpolating gps track to ramses frequency")
+    gps_data = gps_data.sort_values(by="time", ascending=True)
+    gps_data = gps_data.reset_index()
+    datetime_ramses = list(set([time.mktime(datetime.strptime(str(v), "%Y-%m-%d %H:%M:%S").timetuple()) for v in list(df.index)]))
+    gps_datetime = [time.mktime(datetime.strptime(str(v), "%Y-%m-%d %H:%M:%S").timetuple()) for v in list(gps_data["time"])]
+    longitude = list(gps_data["longitude"])
+    latitude = list(gps_data["latitude"])
+    longitude_interp = np.interp(datetime_ramses, gps_datetime, longitude)
+    latitude_interp = np.interp(datetime_ramses, gps_datetime, latitude)
+    df["longitude"] = longitude_interp
+    df["latitude"] = latitude_interp
+
+    # add other ramses data
+    for path in param_paths:
+        if path in spectral_paths:
+                continue
+        if "RAMSES" not in path:
+            print(f"ship data needs to be downloaded separately. Skipping {path}")
+            continue            
+        print(f"Add {path} to spectral profile")    
+        tseries_idx = vessel_paths.index(path)
+        var = vessel_signals[tseries_idx].get_tseries(
+                    pub_tsb,
+                    header=header,
+                    noqc=noqc,
+                    dt=dt,
+                    start_time=start_time,
+                    end_time=end_time,
+            )
+        df = df.merge(var, on="time", how="outer")        
+    return df
+
+
+def get_ramses_data(
+    vessel_name: str,
+    param_paths: list,
+    start_time,
+    end_time,
+    noqc,
+    header,
+    dt=0,
+    pub_tsb=PUB_TSB,
+    meta_host=PUB_META,
+):
+    print("Downloading data for ", vessel_name)
+
+    vessel_signals, vessel_paths = get_paths_measurements(
+        vessel_name, meta_host=meta_host, header=header
+    )
+    spectral_paths = [path for path in param_paths if vessel_signals[vessel_paths.index(path)].TTYPE=="spectra"]
+    assert len(spectral_paths)==1, "Only one spectral profile can be downloaded at a time"
+
+    start_datetime=datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S")
+    end_datetime=datetime.strptime(end_time, "%Y-%m-%dT%H:%M:%S")
+
+    #slide in 12 h slices
+    period =timedelta(hours=12)
+    if (end_datetime-start_datetime)>period:
+        periods = int((end_datetime-start_datetime)/period)
```

Comment:
periods will be undefined if requested range is 12 hours or below so this code fails

```python
def test_get_ramses_data_6h():
    header = token2header("tests/niva-service-account.json")
    start_time = "2023-08-01T12:00:00"
    end_time = "2023-08-01T18:00:00"


    ts = get_ramses_data(
        vessel_name="FA",
        param_paths=["FA/RAMSES/RADIANCE/LD/CALIBRATED"],
        start_time=start_time,
        end_time=end_time,
        noqc=False,
        header=header,
        dt=0,
        pub_tsb="https://ferrybox.p.niva.no/v1/tsb",
        meta_host="https://ferrybox.p.niva.no/v1/metaflow"
    )


    assert not ts.empty
```